cmd/acb/commands/exec/exec.go

```diff
@@ -201,55 +199,49 @@ var Command = cli.Command{
 			}
 		}
 
+		// Add all creds provided by the user in the --credential flag
+		credentials, err := graph.CreateRegistryCredentialFromList(creds)
+		if err != nil {
+			return err
+		}
+
 		versionInUse := graph.FindVersion(template.GetData())
 		shouldIncludeAlias := versionInUse == "" || versionInUse >= "v1.1.0"
-
+		var alias *graph.Alias
 		var task *graph.Task
+		// TODO: (sam) refactor this.
+		// Find a way to do "rendering" and "preprocessing" in Task initialization `graph` module, instead of `commands`.
 		if shouldIncludeAlias {
 			log.Printf("Alias support enabled for version >= 1.1.0, please see https://aka.ms/acr/tasks/task-aliases for more information.")
-			// Generate the base task file without resolving environment variables.
-			task, err = graph.NewTaskFromBytes(template.GetData(), true)
-			if err != nil {
-				return err
+			// Preprocess the task to replace all aliases based on the alias sources.
+			post, aliasPtr, _, aliasErr := graph.PreprocessBytes(template.GetData())
+			alias = aliasPtr
```

Comment:
@djyou @juliusl @mnltejaswini  golang experts, Question about scope :
using 
```
post, alias, _, aliasErr := graph.PreprocessBytes(template.GetData())
```

would give me a compile error `alias declared and not used`

which is why this extra assignment. LMK if theres some idiomatic way out there :)

Reply:
For the particular issue, you can declare other variables outside if block along with alias. So you don't need to use := and avoid variable shadowing.

---
In reply to: [326838074](https://github.com/Azure/acr-builder/pull/510#discussion_r326838074) [](ancestors = 326838074)